### PR TITLE
Implement Table::ReadRow()

### LINKED ISSUE #29: Implement (sync) ReadRow() a read for a single row
At the end of this task the `bigtable::Table` class will have a member function to read a single row, and the unit tests associated with it.
The member function will use the existing policies to determine whether to retry any RPC requests, and how to backoff the RPC requests.
The member function will return a `Row` class with all the data in memory.
This function should take both a `absl::string_view` parameter for the row key and a `bigtable::Filter` parameter to control the filter expression associated with the request.

Consider splitting this issue if the`Filter` class is not ready.

### PR DESCRIPTION
**Pull request by coryan**
This fixes #29.

### CODE REVIEW

File: bigtable/client/table_readrow_test.cc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/table.h"
+#include "bigtable/client/testing/table_test_fixture.h"
+
+namespace {
+class MockResponseStream : public grpc::ClientReaderInterface<
```
Comment (DanielMahu) [inline bigtable/client/table_readrow_test.cc]: maybe move to table_test_fixture.h, since it's the same with the one in table_readrows_test?

File: bigtable/client/table_readrow_test.cc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        *r = response;
+        return true;
+      }))
+      .WillOnce(Return(false));
+  EXPECT_CALL(*stream, Finish()).WillOnce(Return(grpc::Status::OK));
+
+  EXPECT_CALL(*bigtable_stub_, ReadRowsRaw(_, _))
+      .WillOnce(Invoke([&stream, this](grpc::ClientContext *,
+                                       btproto::ReadRowsRequest const &req) {
+        EXPECT_EQ(1, req.rows().row_keys_size());
```
Comment (DanielMahu) [inline bigtable/client/table_readrow_test.cc]: Nice, I didn't realize the EXPECT_* macros work in lambdas as well. I wrote some silly matchers in my tests.

File: bigtable/client/row_reader_test.cc [hunk 3]
```diff
@@ -159,6 +159,7 @@ class RowReaderTest : public bigtable::testing::TableTestFixture {
 TEST_F(RowReaderTest, EmptyReaderHasNoRows) {
   EXPECT_CALL(*bigtable_stub_, ReadRowsRaw(_, _)).WillOnce(Return(stream_));
   EXPECT_CALL(*stream_, Read(_)).WillOnce(Return(false));
+  EXPECT_CALL(*stream_, Finish()).WillOnce(Return(grpc::Status::OK));
```
Comment (DanielMahu) [inline bigtable/client/row_reader_test.cc]: What happens (in general) if client code does _not_ call Finish? Couldn't find an answer to this in the gRPC documentation:
https://grpc.io/grpc/cpp/classgrpc_1_1internal_1_1_client_streaming_interface.html
but if the answer is: a call is left dangling, then I should take care of this in the RowReader

Comment (coryan) [reply bigtable/client/row_reader_test.cc]: The gRPC documentation is awful about these things.  I would expect that not calling `Finish()` leaks whatever resources are needed to return the result of `Finish()` when it is eventually called, right? 

On a separate (but related note) I have observed assertion errors for any streaming gRPC that does not follow exactly the right sequence of calls:

https://github.com/GoogleCloudPlatform/cpp-docs-samples/issues/16

In short, I think if we step outside the **exact** sequence of calls that gRPC expects we are wondering into undefined behavior territory.

File: bigtable/client/table_readrow_test.cc (hunk 2 above)
Comment (coryan) [reply bigtable/client/table_readrow_test.cc]: `EXPECT_EQ()` works fine inside lambdas.  `ASSERT_EQ()` does not always work because it expands to `if (blah) { return; }`, and the lambda may return something other than `void`.  `googletest` should use exceptions :grin:

File: bigtable/client/table_readrow_test.cc (hunk 1 above)
Comment (coryan) [reply bigtable/client/table_readrow_test.cc]: Good idea.  Will refactor before merging.
